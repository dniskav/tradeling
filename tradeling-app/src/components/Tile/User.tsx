import React from 'react';

interface ITileUser {
  avatar_url: string;
  html_url: string;
  name: string;
  location: string;
}

const TileUser: React.FC<ITileUser> = ({ avatar_url, html_url, name, location }) => {
  return(
    <div className="tile">
      <div className="img-container">
        <img src={`${avatar_url}&size=120`} alt={name} />
      </div>
      <div  className="description">
        <h3>
          <a href={html_url}>{name}</a>
        </h3>
        <span>{location}</span>
      </div>
    </div>
  )
};

export default TileUser;
