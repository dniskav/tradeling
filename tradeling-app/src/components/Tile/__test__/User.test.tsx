import User from "../User";
import { shallow } from "enzyme";

describe("<Results />", () => {
  const props = {
    avatar_url: '',
    html_url: '',
    location: '',
    url: '',
    name: '',
    language: '',
    stars: '',
    owner: '',
  }
  const tree = shallow(<User {...props}/>);
  it("Search should match snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});