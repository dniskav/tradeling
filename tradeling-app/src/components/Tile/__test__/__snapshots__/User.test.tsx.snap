// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Results /> Search should match snapshot 1`] = `
<div
  className="tile"
>
  <div
    className="img-container"
  >
    <img
      alt=""
      src="&size=120"
    />
  </div>
  <div
    className="description"
  >
    <h3>
      <a
        href=""
      />
    </h3>
    <span />
  </div>
</div>
`;
